src/raiden_libs/service_registry.py

```diff
@@ -176,15 +190,140 @@ def register(
     )
 
 
-# Seperate function to make testing easier
+def get_token_formatter(
+    token_contract: Contract, min_sig_figures: int = 3
+) -> Callable[[float], str]:
+    """
+    Return a function to nicely format token amounts.
+
+    It will round to full tokens according to the decimal definition in the
+    token contract, but it will show at least `min_sig_figures` significant
+    figures. This ensures that small token amount stay readable and won't get
+    rounded to zero.
+    """
+    symbol = token_contract.functions.symbol().call()
+    decimals = token_contract.functions.decimals().call()
+
+    def format_token_amount(amount: float) -> str:
+        if amount == 0:
+            return f"0 {symbol}"
+        amount /= 10 ** decimals
+        show_dec = max(-floor(log10(abs(amount)) + 1) + min_sig_figures, 0)
+        return ("{:." + str(show_dec) + "f} {}").format(amount, symbol)
+
+    return format_token_amount
+
+
+def send_registration_transaction(
+    web3: Web3,
+    service_registry_contract: Contract,
+    deposit_token_contract: Contract,
+    maybe_prompt: Callable,
+    account_balance: int,
+    service_address: Address,
+) -> None:
+    # Get required deposit
+    required_deposit = service_registry_contract.functions.currentPrice().call()
+    click.secho(
+        f"\nThe current required deposit is fmt_amount({required_deposit})"
+        "\n\tNote: The required deposit continuously decreases, but "
+        "\n\t      increases significantly after a deposit is made."
+    )
+    if web3.eth.chainId == 1:
+        click.secho("You don't have sufficient tokens", err=True)
+        sys.exit(1)
```

Comment:
I don't understand the flow here:
a) where is the check that the user does not have "sufficient tokens"?
b) why don't we show the token amount the user has?

Reply:
You are completely right. I moved those lines around a few times too often and apparently mixed up the conditions the last time. Should be better, now.